src/cli_util.rs

```diff
@@ -1007,6 +1012,23 @@ fn resolve_single_op_from_store(
     }
 }
 
+fn load_revset_aliases(ui: &mut Ui) -> Result<RevsetAliasesMap, CommandError> {
+    const TABLE_KEY: &str = "revset-alias";
+    let mut aliases_map = RevsetAliasesMap::new();
+    if let Ok(table) = ui.settings().config().get_table(TABLE_KEY) {
+        for (decl, value) in table.into_iter().sorted_by(|a, b| a.0.cmp(&b.0)) {
+            let r = value
+                .into_string()
+                .map_err(|e| e.to_string())
+                .and_then(|v| aliases_map.insert(&decl, v).map_err(|e| e.to_string()));
```

Comment:
What do you think about aliases called e.g. `root` or `@`? Should we allow them? It gets trickier because we often use `@` in defaults, so defining it to mean something else will probably have very weird consequences. How about names conflicting with branch names? We could require parentheses after aliases to reduce the risk of conflicts, but that also makes them harder to use. I know someone from the Mercurial community (they don't seem to be on GitHub) thought it was a mistake to allow revset aliases without parentheses. Thoughts?

Reply:
I personally have no problem with Mercurial's revset alias which precedes any valid symbols. Checking conflicts with branch names, etc. is fragile because branches are glowing list. So it's better to require explicit notation if we want to avoid name conflicts. OTOH, there aren't many ASCII characters which can be used without quotes.

Reply:
I haven't had any problems with it either. We actually also have a few revset aliases at Google and we tell users to use (e.g. `smart` for our smartlog equivalent) and I don't think we've heard any reports of users accidentally defining something that overrides another symbol. If we were to disambiguate them in some way, my preference would be to do it using function syntax (i.e. allowing *only* revset alias functions).

Reply:
> If we were to disambiguate them in some way, my preference would be to do it using function syntax (i.e. allowing only revset alias functions).

Totally agree about that.

Even function alias can override the builtins, I feel it's less surprising as the syntax looks doing something more than trivial symbol resolution.

Reply:
Marked as resolved assuming there's no strong concern about symbol shadowing. If that becomes a real problem, we can disable the symbol aliases.

Reply:
Yep, sounds good!